### ISSUE TITLE
Dense ingest never releases CUDA VRAM cache -> long-lived GPU encode climbs to the card ceiling

### ISSUE THREAD

**Issue by mbachaud** [SwiftWing21/helix-context#176]
## Summary

helix's dense encoder never releases torch's CUDA caching-allocator cache, so any long-lived process doing **GPU** dense ingest climbs to the card's VRAM ceiling and then spills to shared system memory (the slow "thrash" that looks like a hang). `v0.6.3` ships dense **ON by default**, so this is the production default retrieval path.

## Evidence

- Surfaced on the new ContextBench **code** benchmark, frozen **v0.6.3** wheel, **1 dense worker** (this is *not* a multi-worker OOM).
- Single-worker dense ingest of one mid-size repo (sympy / scikit-learn, ~1.1k–1.4k files) climbed to **~11.7 GB / 95%** of a 12 GB RTX 3080 Ti, and kept climbing **within a single task**.
- `grep -rn "empty_cache\|ipc_collect\|PYTORCH_CUDA_ALLOC" helix_context/` → **0 hits**.

## Root cause

torch's CUDA caching allocator keeps a **separate cached block per distinct input shape**. The ingest path batch-encodes passages of widely varying length, so each new shape adds a cached block and the cache grows monotonically across a long run; it is never returned to the OS. On Windows/WDDM, once dedicated VRAM fills the driver spills to shared system RAM — the slow path that presents as a multi-hour hang.

## Impact (GPU dense paths)

- daemon `/ingest` route (long-lived server process)
- `scripts/backfill_bgem3_v2.py`
- large genome builds (e.g. the 100k+-file / 500K-corpus builds)
- any bench/tooling that forces dense onto CUDA

CPU dense encoding is unaffected (no CUDA cache). This is a **distinct axis** from the daemon's *system-RAM* commit ceiling at 100+ shards — that one is host RAM, this one is VRAM.

## Fix

Periodic `torch.cuda.empty_cache()` in `BGEM3Codec.encode_batch`, **cuda-gated**, env-tunable `HELIX_DENSE_VRAM_RELEASE_EVERY` (default `256`, `0` disables). `empty_cache()` frees only *unused* cached blocks, so emitted vectors are byte-identical. Measured effect: **~11.7 GB climb → ~6 GB plateau**.

Also recommend documenting `PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True` — it eliminates fragmentation and on its own brought the *starting* peak from 11.7 → 7.7 GB (but did not stop the within-task climb; the periodic release does).

## Notes

Found during the ContextBench code-retrieval track (laude). Flagging for review by the dense-path owner — PR attached.
